sql/updates/reverse-dev.sql

```diff
@@ -0,0 +1,6 @@
+DROP FUNCTION IF EXISTS  @extschema@.add_retention_policy(REGCLASS, "any", BOOL, INTERVAL);
+CREATE FUNCTION @extschema@.add_retention_policy(relation REGCLASS, drop_after "any", if_not_exists BOOL = false)
+RETURNS INTEGER AS '@MODULE_PATHNAME@', 'ts_policy_retention_add' LANGUAGE C VOLATILE;
+DROP FUNCTION IF EXISTS  @extschema@.add_compression_policy(REGCLASS, "any", BOOL, INTERVAL);
+CREATE FUNCTION @extschema@.add_compression_policy(hypertable REGCLASS, compress_after "any", if_not_exists BOOL = false)
+RETURNS INTEGER AS '@MODULE_PATHNAME@', 'ts_policy_compression_add' LANGUAGE C VOLATILE;
```

Comment:
```suggestion
DROP FUNCTION IF EXISTS  @extschema@.add_retention_policy(REGCLASS, "any", BOOL, INTERVAL);
DROP FUNCTION IF EXISTS  @extschema@.add_compression_policy(REGCLASS, "any", BOOL, INTERVAL);
```

You don't need CREATE FUNCTION here as well.. it will be automatically generated as well.

Reply:
It seems the `CREATE FUNCTION` is needed here actually.. From what I understand, the “rule” is, we don’t need `create function` in either `reverse-dev` or `latest-dev` , generally. But if we modify the function signature, as I have done here adding the extra parameter, then we need to create the other version. In this case I need to drop the current version function and then create the other version in  `reverse-dev` since `policy_api.sql` contains the new signature.

Reply:
It actually is needed to not introduce a security vulnerability. In the downgrade script you will have `CREATE` followed by `CREATE OR REPLACE` with the plain `CREATE` a security vulnerability is introduced.